pimd/pim_igmp_mtrace.c

```diff
@@ -28,6 +28,26 @@
 #include "pim_macro.h"
 #include "pim_igmp_mtrace.h"
 
+static struct in_addr mtrace_primary_address(struct interface *ifp)
```

Comment:
Shouldn't this function use pim_ifp->primary_address if it is available else do the lookup below?  Wouldn't that be better?

Reply:
I was doing that before but it turns out that sometimes pim_ifp is not available, when you look up next hop and PIM is not enabled towards the next hop. I could change the code to use pim_ifp->primary_address when it is available...

Reply:
Yes that is exactly what I am suggesting, use pim_ifp->primary_address if it is available else look it up.